### ISSUE TITLE
Feat: add own tabline component separator

### ISSUE THREAD

**Issue by frolvanya** [nvim-lualine/lualine.nvim#1173]
### Requested feature
Right now we can set component separator for both sections and tabline using one option field called `component_separators`. Can you, please, add a new field like `tabline_separators` to provide more customization to users?

Examples:
<img width="1512" alt="image" src="https://github.com/nvim-lualine/lualine.nvim/assets/59515280/05724a55-257e-43e9-b6a4-43ac9b2bc24c">
<img width="1512" alt="image" src="https://github.com/nvim-lualine/lualine.nvim/assets/59515280/dc303795-b6eb-4739-a24f-44de142f5f62">

### Motivation
I'd like to have '|' separator in sections (statusline) and empty separator in tabline

**Comment by daephx** [comment]
I had some trouble modifying the tabline separators independently awhile ago as well. 
In my case, I just wanted something similar to https://github.com/nvim-lualine/lualine.nvim/blob/master/examples/slanted-gaps.lua where the top and bottom are inverted,
but I did not like the gaps between each section.

![image](https://github.com/nvim-lualine/lualine.nvim/assets/36192863/6bef3aa1-8076-46f9-8a84-33615b32d606)

While better control from the component options might be preferable.
Perhaps you can get what you want in the meantime by modifying this function.

I picked out and tweaked this from the above mentioned example theme.
After removing things I didn't need, this is the version I ended up with.

```lua
---Override separators for tabline between components and sections
---@param sections table
---@return table
local function process_sections(sections)
  for _, section in pairs(sections) do
    for id, comp in ipairs(section) do
      if type(comp) ~= "table" then
        comp = { comp }
        section[id] = comp
      end
      -- NOTE: These should only apply to the tabline.
      -- Modify them as you see fit!
      comp.component_separators = { left = "", right = "" }
      comp.section_separators = { left = "", right = "" }
    end
  end
  return sections
end
```

I use lazy.nvim, so I just call it in the config function and pass it the `opts.tabline` before loading lualine.

```lua
config = function(_, opts)
  -- Invert tabline separators before loading plugin
  opts.tabline = process_sections(opts.tabline)
  require("lualine").setup(opts)
end
```

**Comment by mvillafuertem** [comment]
<img width="856" alt="Image" src="https://github.com/user-attachments/assets/b10ca94d-30bb-470c-8c5e-91ed53aa2ca0" />

**Comment by frolvanya** [comment]
> [@frolvanya](https://github.com/frolvanya) Did you find a solution? [@daephx](https://github.com/daephx) The workaround is not working for me.
> 
> <img alt="Image" width="856" src="https://private-user-images.githubusercontent.com/24855057/416633947-b10ca94d-30bb-470c-8c5e-91ed53aa2ca0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA0OTY1MjEsIm5iZiI6MTc0MDQ5NjIyMSwicGF0aCI6Ii8yNDg1NTA1Ny80MTY2MzM5NDctYjEwY2E5NGQtMzBiYi00NzBjLThjNWUtOTFlZDUzYWEyY2EwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI1VDE1MTAyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5OWYxNmQzN2E2M2ZjNTJhYmEyMWYyZGIxNTRhOTJlN2I3ZTA2MTI2NjMzZWU3ZWE3ODI2MzIxMTUzNzZlMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AZ6WF85buEY-PbR2Z4OwMJZ0YvU3Vz9jv_ehrYjKMvE">

Nope, I found that it looks good without separators, so I just forgot about it
